### PR TITLE
fix: Table Attributes expand-row-keys has changed, but fold not work

### LINKED ISSUE #3653: [Bug Report] Table Attributes "expand-row-keys"  has changed, but fold not work.
### Element Plus version
1.0.2-beta.71

### OS/Browsers version
Windows 10  Edge93

### Vue version
3.2.3

### Reproduction Link
https://codesandbox.io/s/element-plus-spa-forked-b6u4z?file=/src/App.vue

### Steps to reproduce
```
<template>
  <div>
    <el-table
      :data="tableData"
      row-key="id"
      border
      :expand-row-keys="expandKeys"
      :tree-props="{ children: 'children', hasChildren: 'hasChildren' }"
    >
      <el-table-column prop="id" label="id"></el-table-column>
      <el-table-column prop="name" label="name"></el-table-column>
    </el-table>
  </div>
  <div>
    <el-button type="primary" @click.stop="handleClick">{{
      expandFlag ? "Fold" : "Expand"
    }}</el-button>
  </div>
</template>


<script>
import { toRefs, reactive } from "vue";
export default {
  setup() {
    const tableData = [
      {
        id: "A1",
        name: "A1",
        children: [
          { id: "A11", name: "A11" },
          { id: "A12", name: "A12" },
        ],
      },
      {
        id: "B1",
        name: "B1",
        children: [
          { id: "B11", name: "B11" },
          { id: "B12", name: "B12" },
          {
            id: "B13",
            name: "B13",
            children: [
              { id: "B131", name: "B131" },
              { id: "B132", name: "B132" },
            ],
          },
        ],
      },
      { id: "C1", name: "C1" },
    ];
    const state = reactive({
      data: [],
      expandKeys: [],
      expandFlag: false,
    });

    const handleClick = () => {
      if (state.expandFlag) {
        // make tree fold
        state.expandFlag = false;
        state.expandKeys = [];
      } else {
        // make tree expand
        state.expandFlag = true;
        state.expandKeys = ["A1", "B1", "B13"];
      }
    };

    return {
      tableData,
      ...toRefs(state),
      handleClick,
    };
  },
};
</script>
```
I cant expanded the tree, but can't fold  by  set "expand-row-key" with empty array.

the element-plus source code bellow
```
  // file: packages\components\table\src\store\tree.ts
  // line: 79
  // method: updateTreeData
  // conent:
      const getExpanded = (oldValue, key) => {
        const included =
          ifExpandAll ||
          (expandRowKeys.value && expandRowKeys.value.indexOf(key) !== -1)
        return !!((oldValue && oldValue.expanded) || included)
      }
```
if `included` was `false`,  but `oldValue.expanded` still `true`, so it return `true`.
这不科学...，希望直接改成  
```
        const included = (expandRowKeys.value && expandRowKeys.value.indexOf(key) !== -1)
        return !!included
```

### What is Expected?
set "expand-row-key" with empty array, and tree folded.

### What is actually happening?
tree still expanded.

### PR DESCRIPTION
**Pull request by Alanscut**
fix: #3653 
My opinion is that if the user sets `expand-row-keys`,  it should overridde `oldValue.expanded`

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/table/src/store/tree.ts [hunk 1]
```diff
@@ -77,10 +77,11 @@ function useTree<T>(watcherData: WatcherPropsData<T>) {
       const oldTreeData = unref(treeData)
       const rootLazyRowKeys = []
       const getExpanded = (oldValue, key) => {
-        const included =
-          ifExpandAll ||
-          (expandRowKeys.value && expandRowKeys.value.indexOf(key) !== -1)
-        return !!((oldValue && oldValue.expanded) || included)
+        if (expandRowKeys.value) {
+          return ifExpandAll || expandRowKeys.value.indexOf(key) !== -1
```
Comment (JeremyWuuuuu) [inline packages/components/table/src/store/tree.ts:81]:
```suggestion
          return ifExpandAll || expandRowKeys.value.includes(key)
```

File: packages/components/table/src/store/tree.ts [hunk 2]
```diff
@@ -77,10 +77,11 @@ function useTree<T>(watcherData: WatcherPropsData<T>) {
...
+        } else {
+          return !!(ifExpandAll || (oldValue && oldValue.expanded))
```
Comment (JeremyWuuuuu) [inline packages/components/table/src/store/tree.ts:83]:
```suggestion
          return !!(ifExpandAll || oldValue?.expanded)
```

**Comment by JeremyWuuuuu** [comment]
cc @Alanscut left a few comments for you, let me know if you have any questions.

**Comment by Alanscut** [comment]
@JeremyWuuuuu  no questions

**Review by jw-foss — APPROVED** [review]

**Pull request merged**